### PR TITLE
Fix fullscreen code block exit control on touch and non-hover devices

### LINKED ISSUE #23535: Expanded code block hides exit control on touch/non-hover devices
**Describe the bug**

When a code block is expanded into fullscreen mode, the exit control is hidden behind hover-only UI.

The fullscreen state can appear to take over the page with no visible way to collapse it again, especially on touch devices or other non-hover contexts.

**To Reproduce**

1. Go to an article with a fenced code block.
2. Click the fullscreen/expand control on the code block.
3. Observe the code block entering fullscreen mode.
4. Try to find a visible collapse/exit control without hovering the code block panel.

**Expected behavior**

Fullscreen code blocks should provide an always-visible, discoverable exit control.

Users should not need hover behavior or prior knowledge of the `Escape` key to leave fullscreen mode.

**Screenshots**

No close button

**Desktop (please complete the following information):**

- OS, version:
- Browser, version:

**Smartphone (please complete the following information):**

- Device: Mac
- OS, version: Tahoe 26.5.1
- Browser, version: Chrome Version 149.0.7827.197 (Official Build) (arm64)

**Additional context**

The current implementation generates both enter and exit icons in `.js-fullscreen-code-action`, then swaps icon visibility in fullscreen mode.

However, `.highlight__panel` is hidden by default and only shown via `.highlight:hover .highlight__panel`, which makes the exit affordance easy to miss or unavailable on touch/non-hover devices.

Related historical context: #8990 introduced the lightbox/fullscreen behavior, but this bug is specifically about exit affordance discoverability after expansion.

Another edge case prevents use of escape key if you were to navigate back > forward in the browser.

- OS, version:
- Browser, version:

**Smartphone (please complete the following information):**

- Device:
- OS, version:
- Browser, version:

**Additional context**

### PR DESCRIPTION
**Pull request by anchildress1**
> [!IMPORTANT]
> This PR was written by Claude (Claude Code, Fable 5 model), directed and reviewed by @anchildress1.
> I told Claude to write "Welcome back Fable!" It replied with
> image

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

The fullscreen code block overlay had no reachable exit on touch/non-hover devices: the exit panel was only revealed on `:hover`, and the overlay rendered *under* the fixed header (`--z-modal: 500` vs `#topbar` at 10000), occluding the control even when visible.

- Exit panel is now always visible in fullscreen; overlay sits above the header (10001, matching the image-lightbox close control)
- Fullscreen state is derived from the DOM (`is-open` class) instead of a module-level boolean, so an interrupted soft navigation can no longer desync clicks from the overlay
- An interrupted session (back navigation while fullscreen) now releases the `overflow: hidden` scroll lock and its listeners

Two small e2e-infra fixes ride along so the new Cypress spec can run: removal of seeds referencing the deleted `ListingCategory` model (crashed all seeded runs with a `NameError`), and a Rails 8 compatibility patch for cypress-rails 0.7 transactions (upstream fix testdouble/cypress-rails#165 is unmerged).

## Related Tickets & Documents

- Closes #23535

## QA Instructions, Screenshots, Recordings

1. Open an article containing a fenced code block
2. Expand the code block to fullscreen (on a touch device or with DevTools touch emulation)
3. The exit control should be visible without hovering and clickable (not buried under the header)
4. Escape key and browser back both exit fullscreen; the page scrolls normally afterward and fullscreen can be re-entered

Tested on macOS: Chrome (Cypress) and jsdom (Jest).

### UI accessibility checklist
- [x] Semantic HTML implemented?
- [x] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

**Before:**
image

**After :**
image

## Added/updated tests?

- [x] Yes (Jest edge cases + a new seeded Cypress flow covering visibility, occlusion, Escape, and back/forward)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
<a href="https://app.blacksmith.sh/forem/codesmith/forem/pr/23543"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1785547014&installation_id=139885019&pr_number=23543&repository=forem%2Fforem&return_to=https%3A%2F%2Fgithub.com%2Fforem%2Fforem%2Fpull%2F23543&signature=e26e7f54a177775cdc300e4d60f58485960218a9e97a51b8f29592714d93b246"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

File: spec/support/seeds/seeds_e2e.rb [hunk 1]
```diff
@@ -909,37 +909,6 @@
...
-    originally_published_at: Time.current,
-    bumped_at: Time.current,
-    tag_list: Tag.order(Arel.sql("RANDOM()")).first(2).pluck(:name),
-  )
-end
-
-##############################################################################
-
 seeder.create_if_none(Tag) do
   10.times do |i|
```
[the remark this answers is not in the document]
Comment (anchildress1) [reply spec/support/seeds/seeds_e2e.rb:913]: Previously removed from `main`. This is just noise that fails tests locally.

File: lib/cypress-rails/manages_transactions.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    def rollback_transaction
+      return if @connection_pools.blank?
+
+      ActiveSupport::Notifications.unsubscribe(@connection_subscriber) if @connection_subscriber
+
+      @connection_pools.each(&:unpin_connection!)
+      @connection_pools.clear
+
+      ActiveRecord::Base.connection_handler.clear_active_connections!
+    end
```
[the remark this answers is not in the document]
Comment (anchildress1) [reply lib/cypress-rails/manages_transactions.rb:51]: Fixed in 87483806d — unsubscribe now runs unconditionally before the blank-pools early return, and the handle is nil'd so repeat rollbacks are inert.

**Review by benhalpern — APPROVED** [review]
Thank you
